packages/mask/src/plugins/Wallet/hooks/useInjectedProviderType.ts

```diff
@@ -0,0 +1,19 @@
+import { useAsyncRetry } from 'react-use'
+import { InjectedProviderType } from '@masknet/web3-shared-evm'
+import { bridgedEthereumProvider } from '@masknet/injected-script'
+import { useUntilEthereumOnline } from './useUntilEthereumOnline'
+
+export function useInjectedProviderType() {
+    const injectedProviderReady = useUntilEthereumOnline()
+    const { value: injectedProviderType = InjectedProviderType.Unknown } = useAsyncRetry(async () => {
+        if (!injectedProviderReady) return InjectedProviderType.Unknown
+        const isC98 = await bridgedEthereumProvider.getProperty('isCoin98')
```

Comment:
You should order those method by frequency. e.g. MetaMask at first

Reply:
Have to do it this way. Cause that providers all declare themself as MetaMask.